plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/task.py

```diff
@@ -0,0 +1,243 @@
+
+            # FlyteSchema
+            if type(dataset) is FlyteSchema:
+                # copy parquet file to user-given directory
+                FlyteContext.current_context().file_access.download_directory(
+                    dataset.remote_path, self._local_file_path
+                )
+
+            # DataFrame (Pandas, Spark, etc.)
+            else:
+                if not os.path.exists(self._local_file_path):
+                    os.makedirs(self._local_file_path, exist_ok=True)
+
+                schema = FlyteSchema(
+                    local_path=self._local_file_path,
+                )
+                writer = schema.open(type(dataset))
+                writer.write(dataset)
+            dataset = os.path.basename(self._local_file_path)
+
+        # minimalistic batch request
+        final_batch_request = {
+            "data_asset_name": dataset,
+            "datasource_name": self._datasource_name,
+            "data_connector_name": self._data_connector_name,
+        }
+
+        # Great Expectations' RuntimeBatchRequest
+        if self._batch_request_config and self._batch_request_config.runtime_parameters:
+            final_batch_request.update(
+                {
+                    "runtime_parameters": self._batch_request_config.runtime_parameters,
+                    "batch_identifiers": self._batch_request_config.batch_identifiers,
+                    "batch_spec_passthrough": self._batch_request_config.batch_spec_passthrough,
+                }
+            )
+
+        # Great Expectations' BatchRequest
+        elif self._batch_request_config:
+            final_batch_request.update(
+                {
+                    "data_connector_query": self._batch_request_config.data_connector_query,
+                    "batch_spec_passthrough": self._batch_request_config.batch_spec_passthrough,
+                }
+            )
+
+        checkpoint_config = {
+            "class_name": "SimpleCheckpoint",
+            "validations": [
+                {
+                    "batch_request": final_batch_request,
+                    "expectation_suite_name": self._expectation_suite_name,
+                }
+            ],
+        }
+
+        if self._checkpoint_params:
+            checkpoint = SimpleCheckpoint(
+                f"_tmp_checkpoint_{self._expectation_suite_name}",
+                context,
+                **checkpoint_config,
+                **self._checkpoint_params,
+            )
+        else:
+            checkpoint = SimpleCheckpoint(
+                f"_tmp_checkpoint_{self._expectation_suite_name}", context, **checkpoint_config
+            )
+
+        # identify every run uniquely
+        run_id = RunIdentifier(
+            **{
+                "run_name": self._datasource_name + "_run",
+                "run_time": datetime.datetime.utcnow(),
+            }
+        )
+
+        checkpoint_result = checkpoint.run(run_id=run_id)
```

Comment:
@talagluck: When trying to validate data against expectations that's bound to fail, I see the following error at this particular line:
```bash
 File "/Users/samhitaalla/.pyenv/versions/3.8.10/lib/python3.8/site-packages/flytekitplugins/great_expectations/task.py", line 224, in execute
    checkpoint_result = checkpoint.run(run_id=run_id)
  File "/Users/samhitaalla/.pyenv/versions/3.8.10/lib/python3.8/site-packages/great_expectations/checkpoint/checkpoint.py", line 742, in run
    return super().run(
  File "/Users/samhitaalla/.pyenv/versions/3.8.10/lib/python3.8/site-packages/great_expectations/checkpoint/checkpoint.py", line 274, in run
    validator: Validator = self.data_context.get_validator(
  File "/Users/samhitaalla/.pyenv/versions/3.8.10/lib/python3.8/site-packages/great_expectations/data_context/data_context.py", line 1778, in get_validator
    batch_definition = batch_list[-1].batch_definition
IndexError: list index out of range
```

Has anything changed in the recent versions? I remember this particular line not throwing an error even if the validation fails. It returns a dictionary that has "success" equal to True or False (let me know if I'm wrong). 

All in all, is this specific error expected?